array_1d overflow warning in StructuralApplication

gcc is giving me this warning when compiling the StructuralApplication, I thought I'd report it since I believe it points to a bug:
```
/home/jcotela/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/yield_surfaces/simo_ju_yield_surface.h: In member function ‘void Kratos::GenericSmallStrainIsotropicDamage<TConstLawIntegratorType>::CalculateMaterialResponseCauchy(Kratos::ConstitutiveLaw::Parameters&) [with TConstLawIntegratorType = Kratos::GenericConstitutiveLawIntegratorDamage<Kratos::SimoJuYieldSurface<Kratos::DruckerPragerPlasticPotential<3> > >]’:
/home/jcotela/Kratos/applications/StructuralMechanicsApplication/custom_constitutive/yield_surfaces/simo_ju_yield_surface.h:143:41: warning: iteration 3 invokes undefined behavior [-Waggressive-loop-optimizations]
             auxf += rStrainVector[cont] * rPredictiveStressVector[cont]; // E:S
```
In fact, `rPredictiveStressVector` is declared as an `array_1d` of size `Dimension`, which I understand to be either 2 or 3, but here it iterates up to position 6.

--- a/applications/StructuralMechanicsApplication/custom_constitutive/yield_surfaces/simo_ju_yield_surface.h
+++ b/applications/StructuralMechanicsApplication/custom_constitutive/yield_surfaces/simo_ju_yield_surface.h
@@ -139,7 +139,7 @@ class SimoJuYieldSurface
         ere1 = SumC / SumA;
 
         double auxf = 0.0;
-        for (std::size_t cont = 0; cont < 6; ++cont) {
+        for (std::size_t cont = 0; cont < VoigtSize; ++cont) {
             auxf += rStrainVector[cont] * rPredictiveStressVector[cont]; // E:S
         }
         rEquivalentStress = std::sqrt(auxf);

